lewis/core/control_server.py

```diff
@@ -78,17 +79,27 @@ def __init__(self, obj, members=None, exclude=None):
 
         self._add_function(':api', self.get_api)
 
-        exposed_members = members if members else self._public_members()
+        exposed_members = members or self._default_members()
 
         for method in exposed_members:
             if not exclude or method not in exclude:
                 self._add_member_wrappers(method)
 
-    def _public_members(self):
+    def _default_members(self):
         """
-        Returns a list of members that do not start with an underscore.
+        Returns a list of members that are good default candidates for being exposed.
+
+        This means public members that appear to be user-created and not part of the framework.
         """
-        return [prop for prop in dir(self._object) if not prop.startswith('_')]
+        def is_valid(prop):
+            return not any([
+                prop.startswith('_'),
+                prop == 'log',
+                prop in dir(Device),
+                prop in dir(StateMachineDevice)
```

Comment:
I don't think getting the device bases classes and `'log'` in here is necessary, it just makes the `control_server` depend on this one very specific thing. I played around a bit and I think it can be made entirely general, what do you think about this:

```python
    def __init__(self, obj, members=None, exclude=None, expose_bases=False):
        super(ExposedObject, self).__init__()

        self._object = obj
        self._function_map = {}

        self._add_function(':api', self.get_api)

        exposed_members = members if members else self._public_members()
        exclude_final = list(exclude or []) + self._get_base_members(
            obj) if not expose_bases else []

        for method in exposed_members:
            if not exclude or method not in exclude_final:
                self._add_member_wrappers(method)

    def _get_base_members(self, obj):
        base_members = []
        for base in inspect.getmro(type(obj))[1:]:
            base_members += dir(base)

        return base_members
```
This automatically catches `'log'` for all subclasses of `DeviceBase` because it's in the dict of that type. For `simulation` and `interfaces` it's necessary to list `'log'` in `Simulation` where they are exposed.